terragrunt/modules/acm-certificate/main.tf

```diff
@@ -0,0 +1,41 @@
+locals {
+  top_level_domains = { for domain in var.domains : domain => join(".", reverse(slice(reverse(split(".", domain)), 0, 2))) }
+}
+
+data "aws_route53_zone" "zones" {
+  for_each = toset(values(local.top_level_domains))
+  name     = each.value
+}
+
+resource "aws_acm_certificate" "cert" {
+  domain_name               = var.domains[0]
+  subject_alternative_names = slice(var.domains, 1, length(var.domains))
+  validation_method         = "DNS"
+
+  lifecycle {
+    create_before_destroy = true
+  }
```

Comment:
The lifecycle is used when changing the certificate (which requires deleting it and creating a new one). Since the certificate is in use (by Cloudfront or a load balancer, depending on where the module is used), we can't delete it before a new one is created, we have to create a new one, switch the resources to use the new one, and then delete the old one. That's what the lifecycle does.